commit 1853576b24ff26b0f4168350435a71b070971ca7
Author: Michael Still <mikal@stillhq.com>
Date:   Mon Jan 19 18:36:15 2026 +1100

    Simplify cron jobs for log rotate.
    
    There was a lot of repetitive jinja for the cron templates for
    logrotate, but they were also fairly infrequently overriden from
    the default. Let's DRY up this code by having a default template
    and only overriding it when required.
    
    Change-Id: I23c3dd66e71a6729a42ea7988931a46ce20d4fc6
    Signed-off-by: Michael Still <mikal@stillhq.com>

diff --git a/ansible/roles/cron/templates/cron-logrotate-aodh.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-aodh.conf.j2
deleted file mode 100644
index fd333f6e4..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-aodh.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/aodh/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-barbican.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-barbican.conf.j2
deleted file mode 100644
index eaa5f0f49..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-barbican.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/barbican/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-blazar.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-blazar.conf.j2
deleted file mode 100644
index 1c0c0e2ca..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-blazar.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/blazar/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-ceilometer.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-ceilometer.conf.j2
deleted file mode 100644
index 5993500a5..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-ceilometer.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/ceilometer/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-cinder.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-cinder.conf.j2
deleted file mode 100644
index b5e59abd2..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-cinder.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/cinder/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-cloudkitty.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-cloudkitty.conf.j2
deleted file mode 100644
index 2dada642d..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-cloudkitty.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/cloudkitty/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-collectd.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-collectd.conf.j2
deleted file mode 100644
index 4e5878be8..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-collectd.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/collectd/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-cyborg.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-cyborg.conf.j2
deleted file mode 100644
index d4f8b9021..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-cyborg.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/cyborg/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-default.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-default.conf.j2
new file mode 100644
index 000000000..1213e5dcb
--- /dev/null
+++ b/ansible/roles/cron/templates/cron-logrotate-default.conf.j2
@@ -0,0 +1,3 @@
+"/var/log/kolla/{{ service }}/*.log"
+{
+}
diff --git a/ansible/roles/cron/templates/cron-logrotate-designate.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-designate.conf.j2
deleted file mode 100644
index c3c475125..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-designate.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/designate/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-etcd.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-etcd.conf.j2
deleted file mode 100644
index 5b009c46e..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-etcd.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/etcd/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-fluentd.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-fluentd.conf.j2
deleted file mode 100644
index 37c76a5f9..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-fluentd.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/fluentd/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-glance-tls-proxy.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-glance-tls-proxy.conf.j2
deleted file mode 100644
index 5f5309918..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-glance-tls-proxy.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/glance-tls-proxy/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-glance.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-glance.conf.j2
deleted file mode 100644
index 5b20985bd..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-glance.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/glance/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-global.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-global.conf.j2
index f57a2e034..74d74b28b 100644
--- a/ansible/roles/cron/templates/cron-logrotate-global.conf.j2
+++ b/ansible/roles/cron/templates/cron-logrotate-global.conf.j2
@@ -19,6 +19,5 @@ maxsize {{ cron_logrotate_log_maxsize }}
 su root kolla
 
 {% for service in cron_logrotate_enabled_services %}
-{% include "cron-logrotate-" ~ service ~ ".conf.j2" %}
-
+{% include ["cron-logrotate-" ~ service ~ ".conf.j2", "cron-logrotate-default.conf.j2"] %}
 {% endfor %}
diff --git a/ansible/roles/cron/templates/cron-logrotate-gnocchi.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-gnocchi.conf.j2
deleted file mode 100644
index d6d5d6f44..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-gnocchi.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/gnocchi/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-grafana.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-grafana.conf.j2
deleted file mode 100644
index 2a3a0889a..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-grafana.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/grafana/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-hacluster.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-hacluster.conf.j2
deleted file mode 100644
index 856193e41..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-hacluster.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/hacluster/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-haproxy.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-haproxy.conf.j2
deleted file mode 100644
index 3e89a2e75..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-haproxy.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/haproxy/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-heat.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-heat.conf.j2
deleted file mode 100644
index 7f5e89a77..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-heat.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/heat/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-horizon.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-horizon.conf.j2
deleted file mode 100644
index f346694f4..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-horizon.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/horizon/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-influxdb.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-influxdb.conf.j2
deleted file mode 100644
index 8a25646d3..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-influxdb.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/influxdb/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-ironic.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-ironic.conf.j2
deleted file mode 100644
index 796c88c3a..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-ironic.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/ironic/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-keystone.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-keystone.conf.j2
deleted file mode 100644
index 79cf10b34..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-keystone.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/keystone/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-kuryr.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-kuryr.conf.j2
deleted file mode 100644
index 3565e1138..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-kuryr.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/kuryr/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-letsencrypt.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-letsencrypt.conf.j2
deleted file mode 100644
index fea08e016..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-letsencrypt.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/letsencrypt/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-magnum.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-magnum.conf.j2
deleted file mode 100644
index 4b2c2af4e..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-magnum.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/magnum/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-manila.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-manila.conf.j2
deleted file mode 100644
index 9b9719514..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-manila.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/manila/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-mariadb.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-mariadb.conf.j2
deleted file mode 100644
index 8543983e0..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-mariadb.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/mariadb/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-masakari.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-masakari.conf.j2
deleted file mode 100644
index 8f71f0975..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-masakari.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/masakari/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-mistral.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-mistral.conf.j2
deleted file mode 100644
index e8e6da8e3..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-mistral.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/mistral/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-nova.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-nova.conf.j2
deleted file mode 100644
index 657a994da..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-nova.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/nova/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-octavia.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-octavia.conf.j2
deleted file mode 100644
index f7464d7e2..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-octavia.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/octavia/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-opensearch.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-opensearch.conf.j2
deleted file mode 100644
index 4f02e14a7..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-opensearch.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/opensearch/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-openvswitch.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-openvswitch.conf.j2
deleted file mode 100644
index 5abab08b2..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-openvswitch.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/openvswitch/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-placement.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-placement.conf.j2
deleted file mode 100644
index 3df479c0d..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-placement.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/placement/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-prometheus.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-prometheus.conf.j2
deleted file mode 100644
index fdfe80714..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-prometheus.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/prometheus/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-proxysql.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-proxysql.conf.j2
deleted file mode 100644
index 7055cce4e..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-proxysql.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/proxysql/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-rabbitmq.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-rabbitmq.conf.j2
deleted file mode 100644
index cf2dc3f36..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-rabbitmq.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/rabbitmq/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-skyline.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-skyline.conf.j2
deleted file mode 100644
index 847870d9a..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-skyline.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/skyline/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-swift.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-swift.conf.j2
deleted file mode 100644
index 0ad2badf1..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-swift.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/swift/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-tacker.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-tacker.conf.j2
deleted file mode 100644
index 681581645..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-tacker.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/tacker/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-trove.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-trove.conf.j2
deleted file mode 100644
index 7a8150c31..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-trove.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/trove/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-valkey.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-valkey.conf.j2
deleted file mode 100644
index f59111e1d..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-valkey.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/valkey/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-watcher.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-watcher.conf.j2
deleted file mode 100644
index e7edaf909..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-watcher.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/watcher/*.log"
-{
-}
diff --git a/ansible/roles/cron/templates/cron-logrotate-zun.conf.j2 b/ansible/roles/cron/templates/cron-logrotate-zun.conf.j2
deleted file mode 100644
index 57eef45c6..000000000
--- a/ansible/roles/cron/templates/cron-logrotate-zun.conf.j2
+++ /dev/null
@@ -1,3 +0,0 @@
-"/var/log/kolla/zun/*.log"
-{
-}
